src/main/scala/com/datastax/sparkstress/ReadTask.scala

```diff
@@ -39,27 +46,124 @@ abstract class ReadTask(config: Config, sc: SparkContext) extends StressTask {
   val coresPerNode:Int = defaultParallelism / numberNodes
   def run()
 
-  def runTrials(sc: SparkContext): Seq[TestResult] = {
+  def runTrials(ss: SparkSession): Seq[TestResult] = {
     println("About to Start Trials")
     for (trial <- 1 to config.trials) yield {
       TestResult(time(run()),0L)
     }
   }
 }
 
+/**
+  * Supertype used to encapsulate reading logic for all dataset read methods.
+  */
+abstract class DatasetReadTask(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+  import org.apache.spark.sql.functions._ // needed to use col()
+  def read_columns(columnNames: Seq[String]): Long = {
+    val columns: Seq[org.apache.spark.sql.Column] = columnNames.map(col(_))
+    config.saveMethod match {
+      // filesystem read methods
+      case "parquet" => ss.read.parquet(s"dsefs:///${keyspace}.${table}").select(columnNames.head, columnNames.tail:_*).count
+      case "text" => ss.read.text(s"dsefs:///${keyspace}.${table}").select(columnNames.head, columnNames.tail:_*).count
+      case "json" => ss.read.json(s"dsefs:///${keyspace}.${table}").select(columnNames.head, columnNames.tail:_*).count
+      case "csv" => ss.read.csv(s"dsefs:///${keyspace}.${table}").select(columnNames.head, columnNames.tail:_*).count
+      // regular read method from DSE/Cassandra
+      case _ => ss
+        .read
+        .format("org.apache.spark.sql.cassandra")
+        .options(Map("table" -> table, "keyspace" -> keyspace))
+        .load()
+        .select(columns:_*)
+        .count()
+    }
+  }
+}
+
+/**
+  * Full Table Scan One Column using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSOneColumn_DS(config: Config, ss: SparkSession) extends DatasetReadTask(config, ss) {
+  override def run(): Unit = {
+    val count = read_columns(Seq("color"))
```

Comment:
for example this request will (and i'm 99% sure on this) get optimized down to

Select (nothing) from Source , which we will turn into SELECT (COUNT(*)) in CQL